File: ecmascript/word-count/word-count.js
class Words {
	constructor() {

	}

	count(words) {
		words = words.toLowerCase();
		const wordCount = {}
		const wordList = words.split(/\s/);
		for (let word of wordList) {
			if (word == '') continue;
			if (wordCount.hasOwnProperty(word)) wordCount[word] += 1;
			else wordCount[word] = 1;
		}
		return wordCount;
	}
}

export default Words;

File: python/word-count/word_count.py
def word_count(txt):
    import re
    word_list = re.split('[\.\,\_\s]', re.sub('[^0-9a-zA-Z\,\_\\s]', '',txt.lower()))
    result = {}
    for word in word_list:
        if word == '':
            continue
        if word in result:
            result[word] += 1
        else:
            result[word] = 1

    return result
    
